core/src/main/java/hudson/util/SecretRewriter.java

```diff
@@ -79,42 +79,36 @@ private String tryRewrite(String s) throws IOException, InvalidKeyException {
     public boolean rewrite(File f, File backup) throws InvalidKeyException, IOException {
         AtomicFileWriter w = new AtomicFileWriter(f, "UTF-8");
         try {
-            PrintWriter out = new PrintWriter(new BufferedWriter(w));
 
             boolean modified = false; // did we actually change anything?
-            try {
-                FileInputStream fin = new FileInputStream(f);
-                try {
+            try (PrintWriter out = new PrintWriter(new BufferedWriter(w))) {
+                try (FileInputStream fin = new FileInputStream(f)) {
```

Comment:
I used intellj automated refactoring here, didn't tried to inspect the code in detail, there's room for further improvements